modules/smtp/modules.php

```diff
@@ -117,13 +117,62 @@ public function process() {
 /**
  * @subpackage smtp/handler
  */
-class Hm_Handler_load_smtp_is_imap_forward extends Hm_Handler_Module
+class Hm_Handler_load_smtp_is_imap_forward_as_attachment extends Hm_Handler_Module
 {
     public function process() {
         if (!$this->module_is_supported('imap')) {
             return;
         }
 
+        if (array_key_exists('forward_as_attachment', $this->request->get)) {
+            $path = explode('_', $this->request->get['list_path']);
+            $imap = Hm_IMAP_List::connect($path[1]);
+            if ($imap && $imap->select_mailbox(hex2bin($path[2]))) {
+                $msg_struct = $imap->get_message_structure($this->request->get['uid']);
+                list($part, $msg_text) = $imap->get_first_message_part($this->request->get['uid'], 'text', 'plain', $msg_struct);
+                $msg_header = $imap->get_message_headers($this->request->get['uid']);
+                if (!array_key_exists('From', $msg_header) || count($msg_header) == 0) {
+                    return;
+                }
+                $content = $imap->get_message_content($this->request->get['uid'], 0, false, false);
+
+                # Attachment Download
+                $attached_files = [];
+                $this->session->set('uploaded_files', array());
```

Comment:
What if there is a draft message in compose page with files attached? Won't we remove those files in that case?

Reply:
I don't understand the question please

Reply:
@kroky, @amaninyumu1 I hope If a draft message is present, we can save its attachments temporarily.

Reply:
transferring a message as an attachment behaves like a new message if you want to save it as a draft you have to click on the save button

Reply:
> if you want to save it as a draft you have to click on the save button

Please confirm if there is an auto-save which does on its own after a while. If not, please create an issue. We auto-save when drafts were saved in the session (https://github.com/cypht-org/cypht/issues/166) but I don't if this was kept when we move to saving drafts in IMAP: https://github.com/cypht-org/cypht/issues/257

Reply:
the logistics is that when you transfer a message as an attachment, it is like a new message, so the automatic saving can no longer be done after a few minutes. you have to click on save if you want to save this message as a draft.

Reply:
Are you sure ? For a new message there is an auto-save which does on its own after a while. I have tested this.... See image below:
<img width="539" alt="autosavedraftnewmessage" src="https://github.com/user-attachments/assets/9f9c493d-a48e-4988-8913-1afc546a8931">

Reply:
Maybe it escaped me. I will try again to confirm. Thank

Reply:
@amaninyumu1 my original question was this - you open a compose form, attach some files (they get stored as uploaded_files in the session), you save this form as a draft and go somewhere else, you then click to forward a message as attachment - this destroys the files you uploaded in a previous draft which seems like a regression in functionality. Maybe at least warn the user that they have unsent email that will be lost?

Reply:
@kroky the attachment files are not saved in the session but rather in the attachment folder dir. So it can't have any impact on other messages. I just tested this confirmation, there was no problem

Reply:
@Baraka24 Thank you very much for your curiosity, autosave works for the functionality that I added. I just tested

Reply:
@amaninyumu1 I don't mean the actual file contents. uploaded_files session key is still used extensively in smtp/modules.php to store references and metadata of attached files from the compose form - drafts, imap messages coming from drafts, etc. If you carelessly clean that session array, all those references are gone even if the file system has the actual files. Honestly, I don't think the attachment file session handling is good enough in Cypht - it is prone to errors with this hard-cleanup even in existing code. One obvious example would be someone opening up 2 compose form tabs and attaching files to both forms. Since this is not part of the original issue, I am going to merge this but we need to improve attachment handling internally.